### ISSUE TITLE
Update email settings UI and handle account reunification

### ISSUE THREAD

**Issue by artemruts** [sourcegraph/sourcegraph#15409]
### Description

To support account reunification when we have multiple "Continue with {codehost}" options,  
we need to provide better support for verifying and setting primary email addresses.  
This in turn will help us make sure users can't have two separate accounts for different code hosts.

### Acceptance Criteria

- [ ] Update current "Emails" view
- [ ] Design updates to list of emails
- [ ] Remove "Mark as unverified" action
- [ ] Add "Primary email address" (Disabled unless the user has more than one verified email) - depends on #15393
- [ ] Enable account unification - depends on #TODO

When user signs in with code host not already associated with an account,  
check the returned email address against verified emails in existing accounts.  
If the account matches, reunify the account.

Account reunification inherently creates a new code host connection.

### Resources

* [✨ Figma](https://www.figma.com/file/zXvqCASHDzmvKbrAl1EVsd/Public-and-Private-Repos-for-Users-on-Cloud?node-id=555%3A220)
* [📃 Productboard](https://sourcegraph.productboard.com/feature-board/2119755-cloud/features/6031053/detail)

**Comment by tsenart** [comment]
@ryanslade: Is the account reunification part of this already done?

**Comment by ryanslade** [comment]
No, no new work has been done on this. I believe we already supported unification between email and GitHub but not GitLab.

**Comment by tsenart** [comment]
> I believe we already supported unification between email and GitHub but not GitLab.

Do we have an issue for that missing feature?

**Comment by ryanslade** [comment]
https://github.com/sourcegraph/sourcegraph/issues/15332
